### ISSUE TITLE
Project Management Workflows

### ISSUE THREAD

**Issue by mattapperson** [Brainfock/Brainfock#4]
So I have been considering the pros/cons of various workflows and such, and stumbled on https://dapulse.com
Their concept for project overview is really good (click to select new status in an overview + flow) vs just a static list or kabana board. I would LOVE to see something like that implemented here!

Also... If you are interested in a larger sponsorship / partnership to push this project along... please hit me up via email

**Comment by Webkadabra** [comment]
@mattapperson hello! Could you please expand a bit more on "click to select new status + flow", meaning - i can't see example you're referring to?
There is definitely a need for a UI/UX to manage existing workflows and topic schemes, I really hope we get to build it :)

**Comment by mattapperson** [comment]
@Webkadabra Sure! You get a screen like this one
<img width="1183" alt="screen shot 2016-06-09 at 8 11 32 am" src="https://cloud.githubusercontent.com/assets/192433/15929233/e23840de-2e19-11e6-84b6-ea107ac736e9.png">
That you can customize the workflow columns to fit your needs (examples: https://dapulse.com/templates/)

Then each column you can click to change the issue status.

This is a fantastic UX for both clients and developers... some major issues exist with it though... it fails however to re-assign issues... the entire dapulse system lacks that concept... it also fails to have related issues, or subtasks... something I am sure you can understand developers need. Not to mention it's not OSS.
If something like the dapulse UX existed, but with the more powerful features Brainfock has... I think it would be a win!

Edit: Note that the assigned columns in this shot are assigned in name only... they are no more assigned the issue then any other subscriber to the issue that comments on a ticket. IMHO a project management system must have assignees to be effective.

**Comment by Webkadabra** [comment]
@mattapperson Thanks for this, I haven't really looked at Dapulse much. Somehow I thought they were opensource.

**Comment by Webkadabra** [comment]
@mattapperson yeah, Dapulse has some common things in overal concept with Brainfock. What is called `templates` in Dapulse are `topic schemes` & `workflows` in Brianfock.

**Comment by mattapperson** [comment]
@Webkadabra I think it also worth looking at clubhouse. Very well built and powerful PM tool without being overly complex

**Comment by mattapperson** [comment]
@Webkadabra Here is my full wish list for Brainfock 😊 
- [ ] Admin interface (in progress)
- [ ] Support something like https://kanbanize.com runtime policies
- [ ]  google drive / dropbox integration
- [ ] Tests coverage
- [ ] VCS integration for software development
- [ ] Support linked issues (for dependencies or blocked issues by other tickets, see https://kanbanize.com for an example of my ideal)
- [ ] Support cross project issue linking (see clubhouse.io)
- [ ] Tagging Tickets
- [ ] Ticket Estimations and time tracking (project total budget and monthly retainer supported)
- [ ] Project overview charts (see dapaulse overview to view milestones)
- [ ] Support issue mirroring between projects (see https://kanbanize.com)

**Comment by Webkadabra** [comment]
@mattapperson this is great, thank You for making that list! We already have couple of guys about to join contributors list, so I will spend some time now to make tickets for items on our roadmap. Some of these items are already in progress, some have UI  ready (linked issues for example), and "activity stream" is what I was going to have next on my plate - there is already a working code for that from PHP version of Brainfock.

It's really nice to see people interested in our project! :+1:

**Comment by mattapperson** [comment]
@Webkadabra would a sponsorship get this stuff accomplished faster?

**Comment by Webkadabra** [comment]
@mattapperson of course! I can't afford work on this project without sponsorship at all (:

**Comment by mattapperson** [comment]
@Webkadabra put some numbers to our needs... ill see what I can do 😊

**Comment by mattapperson** [comment]
@Webkadabra updated my list to make it shorter 😊
